torch/csrc/jit/passes/alias_analysis.h

```diff
@@ -61,9 +61,19 @@ class AliasDb {
   // value in group `b`? i.e. may they overlap?
   TORCH_API bool mayAlias(const ValueSet& a, const ValueSet& b) const;
 
+  // Do any nodes write to an alias set input to `n`?
+  TORCH_API bool hasInputWriters(const Node* n) const;
+
+  // Do any nodes write to an alias set output by `n`?
+  TORCH_API bool hasOutputWriters(const Node* n) const;
+
   // Do any nodes write to an alias set inputed/outputed by `n`?
   TORCH_API bool hasWriters(const Node* n) const;
 
+  // Is the operation in-place? i.e. doesn't write anywhere but locations it
+  // reads from.
+  TORCH_API bool isInPlace(Node* n) const;
```

Comment:
`isMutable` might be better—given the way you have defined "in-place", there is no difference afaict